src/main/java/ca/bc/gov/ag/efax/pdf/service/PdfService.java

```diff
@@ -1,64 +1,82 @@
 package ca.bc.gov.ag.efax.pdf.service;
 
 import java.io.File;
-import java.io.FileOutputStream;
-import java.math.BigDecimal;
+import java.io.IOException;
+import java.net.URL;
+import java.util.Base64;
 
-import org.apache.commons.io.IOUtils;
+import javax.xml.bind.JAXBElement;
+
+import org.apache.commons.io.FileUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.oxm.jaxb.Jaxb2Marshaller;
 import org.springframework.stereotype.Service;
+import org.springframework.ws.client.core.support.WebServiceGatewaySupport;
 
-import ca.bc.gov.ag.efax.pdf.util.PDFParser;
-import ca.bc.gov.ag.outputservice.OutputServiceUtils;
+import ca.bc.gov.ag.efax.pdf.config.PdfProperties;
+import ca.bc.gov.ag.efax.pdf.model.ObjectFactory;
+import ca.bc.gov.ag.efax.pdf.model.PDFTransformations;
+import ca.bc.gov.ag.efax.pdf.model.PDFTransformationsResponse;
+import ca.bc.gov.ag.efax.pdf.util.PdfUtils;
 
 @Service
-public class PdfService {
-
-    private static Logger logger = LoggerFactory.getLogger(PdfService.class);
+@ConditionalOnProperty(
+        name = "aem.enabled",
+        havingValue = "true",
+        matchIfMissing = true)
+public class PdfService extends WebServiceGatewaySupport {
+        
+    private Logger logger = LoggerFactory.getLogger(PdfService.class);
 
-    @Autowired
-    private OutputServiceUtils outputServiceUtils;
+    public PdfService(Jaxb2Marshaller pdfMarshaller, PdfProperties pdfProperties) {        
+        setDefaultUri(pdfProperties.getEndpoint());
+        setMarshaller(pdfMarshaller);
+        setUnmarshaller(pdfMarshaller);
+    }
 
-    public File flattenPdf(String urlString, String fileName) {
-        FileOutputStream fos = null;
+    /**
+     * Attempts to flatten (aka normalize/simplify) a PDF by delegating the call to a web service.
+     * @param url a URL to a PDF to flatten
```

Comment:
isn't this the api gateway endpoint?

Reply:
Not quite.  When JUSTIN sends a fax, the attachment is a URL (not an included attachment) to a PDF doc.  This method will take this file attachment (URL) to a pdf and attempt to flatten it.